extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -442,14 +442,20 @@ export class AmpConsent extends AMP.BaseElement {
     if (!isExperimentOn(this.win, 'amp-consent-geo-override')) {
       return this.getConsentRequiredPromiseLegacy_();
     }
-    if (typeof this.consentConfig_['consentRequired'] === 'boolean') {
-      return Promise.resolve(this.consentConfig_['consentRequired']);
-    } else {
-      return this.getConsentRemote_().then(consentInfo => {
-        const remoteResponse = consentInfo['consentRequired'];
-        return typeof remoteResponse === 'boolean' ? remoteResponse : true;
+    return this.consentStateManager_
+      .getLastConsentInstanceInfo()
+      .then(storedInfo => {
+        if (hasStoredValue(storedInfo)) {
+          return Promise.resolve(false);
+        }
+        if (typeof this.consentConfig_['consentRequired'] === 'boolean') {
+          return Promise.resolve(this.consentConfig_['consentRequired']);
+        }
+        return this.getConsentRemote_().then(consentInfo => {
+          const remoteResponse = consentInfo['consentRequired'];
+          return typeof remoteResponse === 'boolean' ? remoteResponse : true;
```

Comment:
now I think we should just return `!!remoteResponse`. having `null` value handled as `true` is weird.